controllers/finalizers.go

```diff
@@ -302,6 +390,34 @@ func (r *MultiClusterHubReconciler) orphanOwnedMultiClusterEngine(m *operatorsv1
 	return nil
 }
 
+func (r *MultiClusterHubReconciler) orphanOwnedHostedMultiClusterEngine(m *operatorsv1.MultiClusterHub) error {
```

Comment:
We aren't supporting adoption of existing MCE's in hosted mode so we don't want to orphan either.